s2-sdk/src/main/java/s2/types/BasinScope.java

```diff
@@ -0,0 +1,7 @@
+package s2.types;
+
+public enum BasinScope {
+  UNSPECIFIED,
+  AWS_US_EAST_1,
+  UNKNOWN
+}
```

Comment:
Why is there both `UNSPECIFIED` & `UNKNOWN`?

Reply:
I think they both reflect different states. Unspecified is a [valid state](https://github.com/s2-streamstore/s2-protos/blob/main/s2/v1alpha/s2.proto#L132) for a basin scope.

But it could also be the case that a user of the java SDK receives an enum value that it doesn't recognize (because the client hasn't been updated to reflect the latest proto), hence the `UNKNOWN` catch all, which seems better than throwing, or considering the new (explicit) enum value as `UNSPECIFIED`.

Reply:
Similar rationale for StorageClass btw: https://github.com/s2-streamstore/s2-sdk-java/blob/04f81638d82503683e23c5d865ff0eb58a25ef47/s2-sdk/src/main/java/s2/types/StorageClass.java#L4-L5